### PR TITLE
fix: straddle message bug

### LINKED ISSUE #1: Miscalculates number of messages when extended character is split between messages
In the 7 bit alphabet, an extended character uses two characters (the escape character and the character itself). This pair is not allowed to be split across messages, so both the escape character and the character itself get shifted into the next message.

The current implementation doesn't handle this properly, and as a result, may miscalculate the number of messages.

An example SMS would be;

> The quick brown fox jumps over the lazy dog. The quick brown fox jumps over the lazy dog. The quick brown fox jumps over the lazy dog. The quick brown f[x jumps over the lazy dog. The quick brown fox jumps over the lazy dog. The quick brown fox jumps over the lazy dog. The quick brown fox jumps over the.

The '[' at character 152 will become "0x1B[" (2 characters), and become part of the 2nd message, which pushes the last "." at the end of the content into a 3rd message, giving the following 3 messages;

> The quick brown fox jumps over the lazy dog. The quick brown fox jumps over the lazy dog. The quick brown fox jumps over the lazy dog. The quick brown f

> [x jumps over the lazy dog. The quick brown fox jumps over the lazy dog. The quick brown fox jumps over the lazy dog. The quick brown fox jumps over the

>.

The current behaviour is that this will return 2 for the number of parts, but it should be returning 3.

### PR DESCRIPTION
**Pull request by atymic**
Closes #1 

Maybe worth some further tests as well, once merged i'll check the trucate PR works as expected

### CODE REVIEW

**Comment by chrisminett** [comment]
Thanks @atymic ! I'll have a look at this in the next week or so. I've been going through the Phlib repos to upgrade them to minimum PHP 7.3 so I'll do this one next, including adding the CI tests which don't appear to have run here.

**Comment by atymic** [comment]
No worries, it did actually run because i got emails from travis haha.

**Comment by atymic** [comment]
Want me to PR bump to 7.3 and github actions CI?

File: tests/SmsLengthTest.php [hunk 1]
```diff
@@ -84,7 +84,22 @@ public function providerSize()
             [str_repeat("exact•max", 67), 'ucs-2', 603, 9, 603],
 
             // empty
-            ['', '7-bit', 0, 1, 160]
+            ['', '7-bit', 0, 1, 160],
+
+            [
```
Comment (chrisminett) [inline tests/SmsLengthTest.php:89]: Could you add a comment explaining what the test is checking, please? Something which explains the value, like in #1 would be great:

> The '[' at character 152 will become "0x1B[" (2 characters), and become part of the 2nd message, which pushes the last "." at the end of the content into a 3rd message

**Comment by chrisminett** [comment]
@atymic I've done all the code style changes I wanted to get done. Could you please rebase your changes from `main`? I think the only differences would be type declarations and named unit test datasets.

Love the check for the modulo! 🚀

**Comment by atymic** [comment]
See #8 

Someone your changes to the repo broke my local git to the point where I could not push at all! Had to delete fork and re-pr

**Pull request closed without merging**